src/android/com/outsystems/firebase/cloudmessaging/OSFirebaseCloudMessaging.kt

```diff
@@ -200,36 +255,70 @@ class OSFirebaseCloudMessaging : CordovaImplementation() {
         return false
     }
 
-    private fun getBadgeNumber() {
-        controller.getBadgeNumber()
-    }
+    private suspend fun registerDevice(callbackContext: CallbackContext) {
+        flow = MutableSharedFlow(replay = 1)
+
+        // if it doesn't have permission, request it
+        val hasPermission = checkPermission(POST_NOTIFICATIONS_PERMISSION)
```

Comment:
Normally `ContextCompat.checkSelfPermission` is used to determine if a permission has been granted or not. More info: https://developer.android.com/training/permissions/requesting#already-granted

Reply:
This code was taken as-is from the removed `OSNotificationPermissions`. Nevertheless, I had a look at the code behind `checkPermission` and found that it points to [this method](https://github.com/OutSystems/cordova-android/blob/4742358601e892a8c8816ef8b1827dfdf7b3754e/framework/src/org/apache/cordova/CordovaInterfaceImpl.java#L238) which appears to call what you mention. 
Considering the above, is it worth the change?

Reply:
We can leave it as is.